app/services/validateCredentials.js

```diff
@@ -3,15 +3,19 @@ const jwt = require('jwt-simple');
 const models = require('../models/index');
 const errors = require('../errors');
 
-module.exports.signIn = body =>
-  models.user
-    .findOne({
-      where: { email: body.email }
-    })
-    .then(user => {
-      if (!user || !bcrypt.compareSync(body.password, user.password)) {
-        throw errors.invalidCredentials('Invalid credentials, please try again');
-      }
+module.exports.signIn = async body => {
+  const user = await models.user.findOne({
+    where: { email: body.email }
+  });
+  if (!user || !bcrypt.compareSync(body.password, user.password)) {
+    throw errors.invalidCredentials('Invalid credentials, please try again');
+  }
+  const previousToken = await models.session.findOne({ where: { userId: user.id } });
+  if (previousToken) {
+    await previousToken.destroy();
+  }
 
-      return jwt.encode(user.email, process.env.SECRET_KEY);
-    });
+  const newSession = await models.session.create({ userId: user.id });
+  const tokenArray = { token: newSession.id, email: user.email };
+  return jwt.encode(tokenArray, process.env.SECRET_KEY);
```

Comment:
Remember not to call env variables directly. Define them in the `config/index` file first.